compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -822,8 +845,8 @@ private BLangLambdaFunction createFilterLambda(Location pos) {
      * @return created lambda function.
      */
     private BLangLambdaFunction createActionLambda(Location pos) {
-        // returns ()
-        BLangValueType returnType = getNilTypeNode();
+        // returns any|error
+        BLangUnionTypeNode returnType = getAnyErrorNilTypeNode();
```

Comment:
`any` includes `nil` right? Do we need to include `Nil` in `getAnyErrorNilTypeNode`? 
Also the `getAnyErrorNilTypeNode()` specifically adds the `nilType` when creating the union, which I think is not required.

Reply:
This is an existing method. I just reused it to get `any|error` type. Will update the method

Reply:
Updated in a8c44c045ac7488c7677972fc828e27e3a229a15